### DISCUSSION TITLE
`</#rrggbb>` is converted to `<!--#rrggbb-->`. What causes it?

### DISCUSSION THREAD

**Discussion by Andre601** [mkdocs/mkdocs#3537] [Q&A]
I noticed that for some reason `</#rrggbb>` is being converted to `<!--#rrggbb-->` which is something I do not want to happen.

Is this something the Python Markdown library causes? Or rather an extension? If it is the former, how can I disable this?
And if the later, which one could it be? MD in HTML?

**Comment by mondeja** [comment]
Had you tried to escape it as HTML?

**Reply by Andre601** [reply]
Doesn't answer my question on what causes it...

Also, how should I HTML escape it when it's inside a inline-codeblock?

**Reply by mondeja** [reply]
I think that Python-Markdown interprets `</#rrggbb>` as an invalid HTML closing tag, but is not clear. See:

```python3
>>> import markdown
>>> markdown.markdown("# Hello\n\n</#rrggbb>")
<h1>Hello</h1>\n<!--#rrggbb-->
>>> markdown.markdown("# Hello\n\n<#rrggbb></#rrggbb>")
<h1>Hello</h1>\n<p>&lt;#rrggbb&gt;<!--#rrggbb--></p>
```

This should be because `</#rrggbb>` is invalid as HTML tag, they must only contains characters `[0-9a-zA-Z]` following the CSS spec. But could be considered as a bug on Python-Markdown because the library is accepting the start tag as valid content escaping it.

**Reply by Andre601** [reply]
That won't work since it's in an inline code block... And Python Markdown will treat it as literal.

Perhaps I can report this to the repo and it may get fixed or a solution is given.

**Reply by Andre601** [reply]
Another issue is, that this - if it would actually work - would break inline code highlighting that I'm using here...

I've created an issue on the Markdown repository. Hopefully there is a fix for this.

**Reply by mondeja** [reply]
I can't reproduce it:

```python
>>> import markdown
>>> markdown.markdown("# Hello\n\nHi `</#rrggbb>` bye")
'<h1>Hello</h1>\n<p>Hi <code>&lt;!--#rrggbb--&gt;</code> bye</p>'
```

Would you mind to provide additional context?

As a workaround you can use `<code>&lt;#rrggbb&gt;</code>` directly, but it is not clear that this is really happening.

**Reply by Andre601** [reply]
I mean, the issue is right there, which is, that it still gives it as `!--#rrggbb--` which is not what I would like.

I just want `</#rrggbb>` as-is without markdown breaking it like that.

**Reply by mondeja** [reply]
Ah OK, I see.
